src/ComponentBase.ts

```diff
@@ -142,7 +149,7 @@ abstract class ComponentBase<P extends React.Props<any>, S extends Object> exten
         if (!Options.shouldComponentUpdateComparator(this.props, nextProps)) {
             let newState = this._buildStateWithAutoSubscriptions(nextProps, false);
             if (newState && !_.isEmpty(newState)) {
-                this.setState(newState);
+                this.setState(newState as Pick<S, K>);
```

Comment:
You should probably cast to `Pick<S, keyof S>`. Or just fix the type signiture of `_buildStateWithAutoSubscriptions`: it is private so that should not hurt anything. Someday we might change _buildState's signature, and then we won't need these casts.